### PR TITLE
Reconcile config on template change

### PR DESCRIPTION
**Pull request by jakub-dzon**
This PR finalizes `OSBuildConfigTemplate` support  by adding following changes:
 - new flag `OSBuildConfig.spec.triggers.templateConfigChange` for specifying whether associated template change should trigger build creation;
 - updating `OSBuildConfig` with latest version of associated `OSBuildConfigTemplate` when template changes.

### CODE REVIEW

File: internal/predicates/osbuildconfig.go [hunk 1]
```diff
@@ -29,5 +29,9 @@ func (OSBuildConfigChangedPredicate) Update(e event.UpdateEvent) bool {
 	if newConfig.Status.LastTemplateResourceVersion != nil && newConfig.Status.CurrentTemplateResourceVersion != nil {
 		templateChanged = *newConfig.Status.LastTemplateResourceVersion != *newConfig.Status.CurrentTemplateResourceVersion
 	}
+
+	templateTriggerEnabled := newConfig.Spec.Triggers.TemplateConfigChange == nil || *newConfig.Spec.Triggers.TemplateConfigChange
```
Comment (ygalblum) [inline internal/predicates/osbuildconfig.go:33]: If we're checking if template change should trigger a new build, we should also check if a new config should trigger a new build (for the generationChanged flag)

File: internal/predicates/osbuildconfig_test.go [hunk 2]
```diff
@@ -49,6 +52,23 @@ var _ = Describe("OSBuildConfig reconciliation predicate", func() {
 				},
 			},
 		),
+		Entry("when template version changed and template trigger is explicitly enabled",
```
Comment (ygalblum) [inline internal/predicates/osbuildconfig_test.go:55]: Should we also check when the template trigger is not set (default behavior)?

Comment (jakub-dzon) [reply internal/predicates/osbuildconfig_test.go:55]: It's covered here: https://github.com/project-flotta/osbuild-operator/pull/63/files#diff-4d944c176d5a7c6d7e5a0e1c0bd2ba6bd49d8e0479a6d8f225840691d81b8ffaR35

File: internal/predicates/osbuildconfig.go (hunk 1 above)
Comment (jakub-dzon) [reply internal/predicates/osbuildconfig.go:33]: :+1:

Comment (jakub-dzon) [reply internal/predicates/osbuildconfig.go:36]: Done

**Review by ygalblum — APPROVED** [review]

**Pull request merged**